auth-web/src/components/pay/eft/ShortNameTransactions.vue

```diff
@@ -0,0 +1,168 @@
+<template>
+  <v-card>
+    <v-card-title class="card-title">
+      <v-icon
+        class="pr-5"
+        color="link"
+        left
+      >
+        mdi-format-list-bulleted
+      </v-icon>
+      <b>Payments Received from {{ state.shortName?.shortName }} ({{ state.totalResults }})</b>
+    </v-card-title>
+    <BaseVDataTable
+      id="eft-transactions-table"
+      class="transaction-list"
+      itemKey="id"
+      :loading="state.loading"
+      loadingText="Loading Transaction Records..."
+      noDataText="No Transaction Records"
+      :setItems="state.results"
+      :setHeaders="headers"
+      :setTableDataOptions="tableDataOptions"
+      :totalItems="state.totalResults"
+      :filters="state.filters"
+      :updateFilter="updateFilter"
+      :pageHide="true"
+      @update-table-options="tableDataOptions = $event"
+    >
+      <template #header-filter-slot />
+      <template #item-slot-transactionDate="{ item }">
+        <span>{{ formatDate(item.transactionDate, 'MMMM DD, YYYY') }}</span>
+      </template>
+      <template #item-slot-depositAmount="{ item }">
+        <span>{{ formatCurrency(item.depositAmount) }}</span>
+      </template>
+    </BaseVDataTable>
+  </v-card>
+</template>
+<script lang="ts">
+import { Ref, defineComponent, reactive, ref, watch } from '@vue/composition-api'
+import { BaseVDataTable } from '@/components'
+import CommonUtils from '@/util/common-util'
+import { DEFAULT_DATA_OPTIONS } from '../../datatable/resources'
+import { DataOptions } from 'vuetify'
+import { EFTTransactionFilterParams } from '@/models/eft-transaction'
+import PaymentService from '@/services/payment.services'
+import _ from 'lodash'
+
+export default defineComponent({
+  name: 'ShortNameTransactions',
+  components: { BaseVDataTable },
+  props: {
+    shortName: {
+      type: Object,
+      default: () => ({
+        shortName: null,
+        transactionDate: null
+      })
+    }
+  },
+  setup (props) {
+    const headers = [
+      {
+        col: 'transactionDate',
+        hasFilter: false,
+        minWidth: '125px',
+        value: 'Payment Received Date'
+      },
+      {
+        col: 'depositAmount',
+        hasFilter: false,
+        minWidth: '125px',
+        value: 'Amount'
+      }
+    ]
+
+    const state = reactive({
+      results: [],
+      totalResults: 1,
+      filters: {
+        isActive: false,
+        pageNumber: 1,
+        pageLimit: 5
+      } as EFTTransactionFilterParams,
```

Comment:
if you build a state object, shouldn't need this casting

Reply:
Changed to state object